pkg/loop/context.go

```diff
@@ -0,0 +1,48 @@
+package loop
+
+import "context"
+
+type ctxKey int
+
+const (
+	ctxKeyJobID ctxKey = iota
+	ctxKeyJobName
+	ctxKeyContractID
+	ctxKeyFeedID
+)
+
+// ContextValues is a helper for passing values via a [context.Context].
```

Comment:
passing values between what? is this across the plugin boundary? is this indirection/obfuscation needed?

Reply:
It is used across the plugin boundary, but not always/only. This was introduced to preserve logger key/vals, since passing a logger as a GRPC service argument doesn't make sense, and the point was just to have the 'request scoped' logger anyways. It is not the only way to achieve this, and it was not strictly necessary, but it keeps things consistent during the transition phase when the feature can be flipped on/off.